resource/config.go

```diff
@@ -172,6 +172,15 @@ func (conf Config) Equals(other Config) bool {
 	other.ImplicitDependsOn = nil
 	other.cachedImplicitDeps = nil
 	other.cachedErr = nil
+
+	// RSDK-4657: Only compare `Config.Attributes` for equality.
+	confConverted, otherConverted := conf.ConvertedAttributes, other.ConvertedAttributes
+	defer func() {
+		conf.ConvertedAttributes, other.ConvertedAttributes = confConverted, otherConverted
```

Comment:
I suppose this is not strictly necessary, but fine to keep in in case someone switches the receiver to be a pointer type.

Reply:
No, you are right. Sorry about this -- you told me a half dozen times these are not pointer receivers and so my code was confusingly defensive. Removing these and documenting that the nil assignments are safe because of these are value types.